pkg/diagnostics/cluster/aggregated_logging/diagnostic.go

```diff
@@ -0,0 +1,207 @@
+		checkDaemonSets(d, d, project)
+		checkServices(d, d, project)
+		checkRoutes(d, d, project)
+		checkKibana(d.result, d.OsClient, d.KubeClient, project)
+	}
+	return d.result
+}
+
+const projectNodeSelectorWarning = `
+The project '%[1]s' was found with a non-empty node selector annotation.  This will keep
+Fluentd from running on certain nodes and collecting logs from the entire cluster.  You
+can correct it by editing the project:
+
+  oc edit namespace %[1]s
+
+and updating the annotation:
+
+  'openshift.io/node-selector' : ""
+
+`
+
+func retrieveLoggingProject(r types.DiagnosticResult, masterCfg *configapi.MasterConfig, osClient *client.Client) string {
+	r.Debug("AGL0010", fmt.Sprintf("masterConfig.AssetConfig.LoggingPublicURL: '%s'", masterCfg.AssetConfig.LoggingPublicURL))
+	projectName := ""
+	if len(masterCfg.AssetConfig.LoggingPublicURL) == 0 {
+		r.Debug("AGL0017", "masterConfig.AssetConfig.LoggingPublicURL is empty")
+		return projectName
+	}
+
+	loggingUrl, err := url.Parse(masterCfg.AssetConfig.LoggingPublicURL)
+	if err != nil {
+		r.Error("AGL0011", err, fmt.Sprintf("Unable to parse the loggingPublicURL from the masterConfig '%s'", masterCfg.AssetConfig.LoggingPublicURL))
+		return projectName
+	}
+
+	routeList, err := osClient.Routes(kapi.NamespaceAll).List(kapi.ListOptions{LabelSelector: loggingSelector.AsSelector()})
+	if err != nil {
+		r.Error("AGL0012", err, fmt.Sprintf("There was an error while trying to find the route associated with '%s' which is probably transient: %s", loggingUrl, err))
+		return projectName
+	}
+
+	for _, route := range routeList.Items {
+		r.Debug("AGL0013", fmt.Sprintf("Comparing URL to route.Spec.Host: %s", route.Spec.Host))
+		if loggingUrl.Host == route.Spec.Host {
+			if len(projectName) == 0 {
+				projectName = route.ObjectMeta.Namespace
+				r.Info("AGL0015", fmt.Sprintf("Found route '%s' matching logging URL '%s' in project: '%s'", route.ObjectMeta.Name, loggingUrl.Host, projectName))
+			} else {
+				r.Warn("AGL0019", nil, fmt.Sprintf("Found additional route '%s' matching logging URL '%s' in project: '%s'.  This could mean you have multiple logging deployments.", route.ObjectMeta.Name, loggingUrl.Host, projectName))
+			}
+		}
+	}
+	if len(projectName) == 0 {
+		message := fmt.Sprintf("Unable to find a route matching the loggingPublicURL defined in the master config '%s'. Check that the URL is correct and aggregated logging is deployed.", loggingUrl)
+		r.Error("AGL0014", errors.New(message), message)
+		return ""
+	}
+	project, err := osClient.Projects().Get(projectName)
+	if err != nil {
+		r.Error("AGL0018", err, fmt.Sprintf("There was an error retrieving project '%s' which is most likely a transient error: %s", projectName, err))
+		return ""
+	}
+	nodeSelector, ok := project.ObjectMeta.Annotations["openshift.io/node-selector"]
+	if ok && len(nodeSelector) != 0 {
+		r.Warn("AGL0030", nil, fmt.Sprintf(projectNodeSelectorWarning, projectName))
+	}
```

Comment:
If the annotation is *not* there then I would like them to get a warning because it leaves the project subject to the master config projectConfig.defaultNodeSelector which could be set now or later to something that would limit fluentd being deployed.